sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -58,6 +61,17 @@ class UDFSuite extends QueryTest {
     ctx.dropTempTable("tmp_table")
   }
 
+  test("SPARK-8005 input_file_name") {
+    withTempPath { dir =>
+      val data = ctx.sparkContext.parallelize(0 to 1000).toDF("id")
```

Comment:
if you specify the number of partitions as 2, i think you can add a test to count the distinct number of paths returned, and then you can check that the number has to >= 2